packages/adblocker-content/adblocker.ts

```diff
@@ -31,16 +31,38 @@ export interface IMessageFromBackground {
   }[];
 }
 
-function debounce<F extends (...args: any[]) => any>(fn: F, waitFor: number) {
+function debounce<F extends (...args: any[]) => any>(
+  fn: F,
+  {
+    waitFor,
+    maxWait,
+  }: {
+    waitFor: number;
+    maxWait: number;
+  },
+) {
   let timeout: NodeJS.Timeout | undefined;
+  let started = -1;
 
-  return (...args: Parameters<F>) => {
-    clearTimeout(timeout);
+  return [
+    (...args: Parameters<F>) => {
+      clearTimeout(timeout);
 
-    timeout = setTimeout(() => {
-      fn(...args);
-    }, waitFor);
-  };
+      if (started > 0 && Date.now() - started >= maxWait) {
+        fn(...args);
```

Comment:
fn would be called here two times (because of the setTimeout afterwards)

Perhaps something like this would work:
```suggestion
        fn(...args);
        started = Date.now(); // or better reuse the timestamp that has been just computed
        return;
```